src/Avalonia.Controls/Control.cs

```diff
@@ -53,24 +55,48 @@ public class Control : InputElement, IControl, INamed, IVisualBrushInitialize, I
         /// Event raised when an element wishes to be scrolled into view.
         /// </summary>
         public static readonly RoutedEvent<RequestBringIntoViewEventArgs> RequestBringIntoViewEvent =
-            RoutedEvent.Register<Control, RequestBringIntoViewEventArgs>("RequestBringIntoView", RoutingStrategies.Bubble);
+            RoutedEvent.Register<Control, RequestBringIntoViewEventArgs>(
+                "RequestBringIntoView",
+                RoutingStrategies.Bubble);
 
         /// <summary>
         /// Provides event data for the <see cref="ContextRequested"/> event.
         /// </summary>
         public static readonly RoutedEvent<ContextRequestedEventArgs> ContextRequestedEvent =
-            RoutedEvent.Register<Control, ContextRequestedEventArgs>(nameof(ContextRequested),
+            RoutedEvent.Register<Control, ContextRequestedEventArgs>(
+                nameof(ContextRequested),
                 RoutingStrategies.Tunnel | RoutingStrategies.Bubble);
         
+        /// <summary>
+        /// Defines the <see cref="Loaded"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> LoadedEvent =
```

Comment:
Do we really want it to be a routed event?
It would cause additional tree traversals for every control which is bad for perf.

Reply:
I followed the WPF and UWP API. Since it is direct it can be a CLR event if needed though. @amwx @maxkatz6 what are your votes for this?

Reply:
From WPF docs "Direct routed events do support other routed event behavior: they support an accessible handlers collection, and can be used as an [EventTrigger](https://docs.microsoft.com/en-us/dotnet/api/system.windows.eventtrigger?view=windowsdesktop-6.0) in a style".
Accessibly handlers part sounds meaningful, but probably not much useful in the case of Loaded event. And we don't have EventTrigger while Behaviors SDK can work with CLR events too.

Reply:
Also note that direct events don't really Traverse the tree more than 1 generation max. So performance concerns are likely not much in this case. However, I did debate using CLR events originally for the added performance benefit as well. In the end decided to follow compatibility. I guess I agree that the original design intent here was probably for event triggers support.

Reply:
Class handlers is another feature of routed events.

Only event handlers on the event source are invoked. This non-routing strategy is analogous to Windows Forms UI framework events, which are standard CLR events. Unlike CLR events, direct routed events support [class handling](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/events/routed-events-overview?view=netdesktop-6.0#class-handlers) and can be used by [EventSetters](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/events/routed-events-overview?view=netdesktop-6.0#eventsetters-and-eventtriggers) and [EventTriggers](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/events/routed-events-overview?view=netdesktop-6.0#eventsetters-and-eventtriggers).

Reply:
> Class handlers is another feature of routed events.

Ok, this one can convince me to keep routed event as we also have class handlers.

Reply:
Ok, I'll keep them routed:

1. This follows the API of WPF and UWP which is good for compatibility
2. It is only a direct routed event so it's really not traversing the tree. Performance impact should be negligible.
3. Routed events support more functionality
   * Class handlers

Reply:
The problem with having `Loaded` as RoutedEvent is that we can't check if there are any subscribers before adding the Visual to the `Loaded` queue. Which means that we are doing extra work for 99.99% of visuals that do not use `Loaded` event for absolutely no reason. Which is a perf regression I'd rather avoid.

Reply:
We don't actually know it is a performance problem yet. I agree it could help speed things up but I'm not sure the sacrifice in functionality and compatibility is worth it.

How about we release the 11.0 preview as-is and see what performance looks like in real app usage? If it's an issue we could make the change.

I dont know performance benchmarking but that could be done as well.

Reply:
Well, please just let me know the core team consensus. Either way will work for my usages and I just hope to close this PR soon.

Reply:
It's definitely easier to add functionality (i.e. make clr event a routed in the future) then remove it (and maybe break some apps).
I will try to run this branch with some of our control benchmarks to see if there is a noticeable difference between clr/routed event.

Reply:
Well, I did benchmark it, and there is a difference, but it close to be a measuring error.

"CreateCalendar" benchmark raises Loaded event 459 times for all kinds of controls. Which means, there are 459 different controls were loaded while Calendar was creating and attached to the root.

Routed event, with queue (last PR commit):
|                             Method |     Mean |    Error |   StdDev |    Gen 0 |    Gen 1 |   Gen 2 | Allocated |
|----------------------------------- |---------:|---------:|---------:|---------:|---------:|--------:|----------:|
| CreateCalendar | 18.69 ms | 0.231 ms | 0.216 ms | 718.7500 | 312.5000 | 62.5000 |      4 MB |
| CreateCalendarWithSubscriber | 18.30 ms | 0.240 ms | 0.224 ms | 718.7500 | 312.5000 | 62.5000 |      4 MB |

Routed event, without queue:
|                             Method |     Mean |    Error |   StdDev |    Gen 0 |    Gen 1 |   Gen 2 | Allocated |
|----------------------------------- |---------:|---------:|---------:|---------:|---------:|--------:|----------:|
|                     CreateCalendar | 18.28 ms | 0.186 ms | 0.165 ms | 718.7500 | 312.5000 | 62.5000 |      4 MB |
| CreateCalendarWithSubscriber | 18.48 ms | 0.181 ms | 0.161 ms | 718.7500 | 312.5000 | 62.5000 |      4 MB |

CLR event, with queue:
|         Method |     Mean |    Error |   StdDev |    Gen 0 |    Gen 1 |   Gen 2 | Allocated |
|--------------- |---------:|---------:|---------:|---------:|---------:|--------:|----------:|
| CreateCalendar | 18.12 ms | 0.357 ms | 0.334 ms | 718.7500 | 312.5000 | 31.2500 |      4 MB |

CLR event, without queue:
|         Method |     Mean |    Error |   StdDev |    Gen 0 |    Gen 1 |   Gen 2 | Allocated |
|--------------- |---------:|---------:|---------:|---------:|---------:|--------:|----------:|
| CreateCalendar | 17.95 ms | 0.319 ms | 0.299 ms | 718.7500 | 312.5000 | 31.2500 |      4 MB |

Reply:
Just an observation: tested with DataGrid page. When there are no DevTools handlers, it still feels fast on my PC. With enabled Loaded and Unloaded even listeners on devtools page it's noticeably slower. But that's expected as for me, since devtools needs to log tons of events in that case.

Reply:
Pushed updated benchmark what I used

Reply:
CC @kekekeks

Reply:
I believe the benchmark doesn't get to the point of raising the event.
```
       public void CreateCalendar()
        {
            var calendar = new Calendar();

            _root.Child = calendar;

            _root.LayoutManager.ExecuteLayoutPass();
        }
```

It just executes a layout pass.

Reply:
Also, you didn't add an optimization that skips the addition to the queue completely if there are no CLR event subscribers.

Reply:
@maxkatz6 Thanks for benchmarking this and testing the alternates!

Reply:
@kekekeks I have pushed code to run dispatcher with loaded priority yesterday. And had double checked if this event actually fired (so I know how many times it did actually fire). 

I haven't added that optimization, but difference between with and without queue still less than measuring error.

Reply:
My instinct is still just to remove the queue as it feels like the much better design. I'm happy to pass this PR off or will make any changes based on your agreement though.